src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -258,10 +260,20 @@ private void addOptions(Template template, Options options, Infrastructure infra
             if (ports != null) {
                 securityGroupName += "-" + UUID.randomUUID();
                 String sgDescription = "Auto generated security group to authorize the ports " + Arrays.toString(ports);
-                AWSSubnetApi awsSubnetApi = getAWSSubnetApi(infrastructure);
-                String vpcId = awsSubnetApi.describeSubnetsInRegion(region, options.getSubnetId()).get(0).getVpcId();
                 CreateSecurityGroupOptions sgOptions = new CreateSecurityGroupOptions();
-                sgOptions.vpcId(vpcId);
+                if (options.getSubnetId() != null) {
+                    AWSSubnetApi awsSubnetApi = getAWSSubnetApi(infrastructure);
+                    FluentIterable<Subnet> subnets = awsSubnetApi.describeSubnetsInRegion(region,
+                                                                                          options.getSubnetId());
+                    if (!subnets.isEmpty()) {
+                        String vpcId = subnets.get(0).getVpcId();
+                        sgOptions.vpcId(vpcId);
+                    } else {
+                        log.warn("The subnet {} could not be described from AWS in region {}. It will be ignored.",
+                                 options.getSubnetId(),
+                                 region);
+                    }
+                }
```

Comment:
`options.getSubnetId()` is used 3 times, may be you can define it in variable just after line 263.

Reply:
Fair enough. Done. Thanks!